### PR TITLE
Remove custom data

### PR DESCRIPTION
**Pull request by mnonnenmacher**
Remove the `data`  property from `OrtResult`, because the `labels` property should now be used for custom data.

### CODE REVIEW

File: model/src/main/kotlin/OrtResult.kt [hunk 1]
```diff
@@ -33,12 +34,11 @@ import org.ossreviewtoolkit.spdx.SpdxExpression
 import org.ossreviewtoolkit.utils.log
 import org.ossreviewtoolkit.utils.zipWithDefault
 
-typealias CustomData = MutableMap<String, Any>
-
 /**
  * The common output format for the analyzer and scanner. It contains information about the scanned repository, and the
  * analyzer and scanner will add their result to it.
  */
+@JsonIgnoreProperties(value = ["data"])
```
Comment (sschuberth) [inline model/src/main/kotlin/OrtResult.kt:41]: Could be just `@JsonIgnoreProperties("data")`.

File: cli/src/main/kotlin/commands/EvaluatorCommand.kt [hunk 2]
```diff
@@ -202,9 +202,7 @@ class EvaluatorCommand : CliktCommand(name = "evaluate", help = "Evaluate rules
 
         outputDir?.let { absoluteOutputDir ->
             // Note: This overwrites any existing EvaluatorRun from the input file.
-            val ortResultOutput = ortResultInput.copy(evaluator = evaluatorRun).apply {
-                data += ortResultInput.data
-            }.mergeLabels(labels)
+            val ortResultOutput = ortResultInput.copy(evaluator = evaluatorRun).mergeLabels(labels)
```
Comment (sschuberth) [inline cli/src/main/kotlin/commands/EvaluatorCommand.kt:205]: Commit message. This should auto-close https://github.com/oss-review-toolkit/ort/issues/2907.

**Review by sschuberth — CHANGES_REQUESTED** [review]

File: model/src/main/kotlin/OrtResult.kt (hunk 1 above)
Comment (mnonnenmacher) [reply model/src/main/kotlin/OrtResult.kt]: Done.

File: cli/src/main/kotlin/commands/EvaluatorCommand.kt (hunk 2 above)
Comment (mnonnenmacher) [reply cli/src/main/kotlin/commands/EvaluatorCommand.kt:205]: Done.

**Review by sschuberth — APPROVED** [review]

**Pull request merged**